pkg/controller/statusmanager/status_manager.go

```diff
@@ -369,12 +369,26 @@ func (status *StatusManager) set(reachedAvailableLevel bool, conditions ...operv
 			)
 		}
 
-		v1helpers.SetOperatorCondition(&oc.Status.Conditions,
-			operv1.OperatorCondition{
-				Type:   operv1.OperatorStatusTypeUpgradeable,
-				Status: operv1.ConditionTrue,
-			},
-		)
+		if oc.Spec.DefaultNetwork.Type == operv1.NetworkTypeOpenShiftSDN {
+			// OpenShiftSDN is removed in 4.17, so block the upgrade if we have OpenShiftSDN in the spec.
+			v1helpers.SetOperatorCondition(&oc.Status.Conditions,
+				operv1.OperatorCondition{
+					Type:   string(configv1.OperatorUpgradeable),
+					Status: operv1.ConditionFalse,
+					Reason: "OpenShiftSDNConfigured",
+					Message: "Cluster is configured with OpenShiftSDN, which is not supported in the next version. Please " +
+						"follow the documented steps to migrate from OpenShiftSDN to OVN-Kubernetes in order to be able to upgrade. " +
+						"https://docs.openshift.com/container-platform/4.15/networking/ovn_kubernetes_network_provider/migrate-from-openshift-sdn.html",
```

Comment:
Also, @mcurry-rh @CFields651, ok on the wording? (This is the message someone would get when trying to upgrade an openshift-sdn cluster from 4.16 to 4.17.)

Reply:
LGTM!

Reply:
> while it might not exist yet, this URL should probably point to 4.16, not 4.15? 
+1

Lets just create a followup card to double check the link works after we have the docs available.

Reply:
done, here is a jira card to track this https://issues.redhat.com/browse/SDN-4784